lib/optimizely/decision_service.rb

```diff
@@ -365,52 +383,49 @@ def get_whitelisted_variation_id(project_config, experiment_key, user_id, decide
 
       whitelisted_variations = project_config.get_whitelisted_variations(experiment_key)
 
-      return nil unless whitelisted_variations
+      return nil, nil unless whitelisted_variations
 
       whitelisted_variation_key = whitelisted_variations[user_id]
 
-      return nil unless whitelisted_variation_key
+      return nil, nil unless whitelisted_variation_key
 
       whitelisted_variation_id = project_config.get_variation_id_from_key(experiment_key, whitelisted_variation_key)
 
       unless whitelisted_variation_id
         message = "User '#{user_id}' is whitelisted into variation '#{whitelisted_variation_key}', which is not in the datafile."
         @logger.log(Logger::INFO, message)
-        decide_reasons&.push(message)
-        return nil
+        return nil, message
```

Comment:
Is this ok to return a string instead of a string array?

Reply:
Yes, if you notice, i am using splat operator `*` when pushing into the array. it flattens the messages if its an array or uses it as is if its a single message.